crates/bevy_pbr/src/material_bind_groups.rs

```diff
@@ -0,0 +1,562 @@
+                                    size: None,
+                                });
+                            }
+                            (
+                                _,
+                                BindingResourceArray::TextureViews(texture_dimension, ref mut vec),
+                            ) => vec.push(&fallback_image.get(texture_dimension).texture_view),
+                            (_, BindingResourceArray::Samplers(ref mut vec)) => {
+                                vec.push(&fallback_bindless_resources.fallback_sampler);
+                            }
+                        }
+                    }
+                }
+
+                Some(ref unprepared_bind_group) => {
+                    // Push the resources for this slot.
+                    //
+                    // All materials in this group must have the same type of
+                    // binding (buffer, texture view, sampler) in each bind
+                    // group entry.
+                    for (&mut (binding, ref mut binding_resource_array), (_, binding_resource)) in
+                        binding_resource_arrays
+                            .iter_mut()
+                            .zip(unprepared_bind_group.bindings.0.iter())
+                    {
+                        match (binding_resource_array, binding_resource) {
+                            (
+                                &mut BindingResourceArray::Buffers(ref mut vec),
+                                OwnedBindingResource::Buffer(buffer),
+                            ) => match NonZero::new(buffer.size()) {
+                                None => vec.push(BufferBinding {
+                                    buffer: &fallback_buffers.0[binding],
+                                    offset: 0,
+                                    size: None,
+                                }),
+                                Some(size) => vec.push(BufferBinding {
+                                    buffer,
+                                    offset: 0,
+                                    size: Some(size),
+                                }),
+                            },
+                            (
+                                &mut BindingResourceArray::TextureViews(_, ref mut vec),
+                                OwnedBindingResource::TextureView(_, texture_view),
+                            ) => vec.push(texture_view),
+                            (
+                                &mut BindingResourceArray::Samplers(ref mut vec),
+                                OwnedBindingResource::Sampler(sampler),
+                            ) => vec.push(sampler),
+                            _ => {
+                                error!(
+                                    "Mismatched bind group layouts; can't combine bind groups \
+                                    into a single bindless bind group!"
+                                );
```

Comment:
Can we add the material type and mismatching resource here? That should speed up debugging quite a bit if this happens.

Reply:
Done. I think what I added is what you wanted.